Source/SwiftLintBuiltInRules/Rules/Performance/EmptyCountRule.swift

```diff
@@ -97,47 +97,24 @@ private extension EmptyCountRule {
         }
 
         override func visit(_ node: InfixOperatorExprSyntax) -> ExprSyntax {
-            guard node.hasBinaryOperator else {
+            guard node.hasBinaryOperator, let binaryOperator = node.binaryOperator else {
                 return super.visit(node)
             }
 
-            if let (count, position) = node.countNodeAndPosition(onlyAfterDot: configuration.onlyAfterDot) {
-                let newNode: ExprSyntax? =
-                    if let count = count.as(MemberAccessExprSyntax.self) {
-                        count.with(\.declName.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
-                    } else if let count = count.as(DeclReferenceExprSyntax.self) {
-                        count.with(\.baseName, "isEmpty").trimmed.as(ExprSyntax.self)
+            if let (count, position) = node.countNodeAndPosition(onlyAfterDot: configuration.onlyAfterDot),
+            let newNode = (
+                count.as(MemberAccessExprSyntax.self)?.with(\.declName.baseName, "isEmpty").as(ExprSyntax.self)
+                ?? count.as(DeclReferenceExprSyntax.self)?.with(\.baseName, "isEmpty").as(ExprSyntax.self)
+            )?.trimmed {
```

Comment:
This is hard to read. Let's have it as a separate statement and check for optionality in another `if` statement thereafter.

Reply:
Fixed at 9965cb0